### DISCUSSION TITLE
Owntracks Recorder

### DISCUSSION THREAD

**Discussion by gene1wood** [Freika/dawarich#408] [Q&A] [answered]
With the conclusion of Google Timeline being accessible outside of one's device, I'm looking to move to an alternative. I see that dawarich can consume data from the OwnTracks mobile app.

I also see that OwnTracks has the [OwnTracks Recorder](https://github.com/owntracks/recorder) server side component that the OwnTracks app can send data to.

How does dawarich differ from OwnTracks Recorder? Are these similar things? If they are, what are the features that dawarich has that makes it better than OwnTracks Recorder?

Context : I haven't used either yet, I've only watched videos showing how they work.

**Comment by Freika** [comment]
In general, they both can visualize location history. There are a number of differences though. Chatgpt helped me build this table, so:

| **Feature**                | **OwnTracks Recorder**                                                                                      | **Dawarich**                                                                                     |
|-----------------------------|------------------------------------------------------------------------------------------------------------|-------------------------------------------------------------------------------------------------|
| **Primary Purpose**         | A self-hosted backend for storing and visualizing location data from the OwnTracks app.                     | An alternative to Google Location History with a focus on privacy and customization.           |
| **Data Source**             | OwnTracks app (publishes location data via MQTT or HTTP).                                                  | Can import data from various sources, such as Google Location History, or receive data from OwnTracks, Overland and GPSLogger mobile aps.  |
| **Visualization**           | Provides basic visualization with a map interface.                                                        | Offers advanced map visualization (e.g., points simplification, fog of war mode, number of map styles).|
| **Extensibility**           | Limited extensibility; primarily designed to work with OwnTracks data.                                     | Open-ended and customizable for various use cases beyond just OwnTracks compatibility.         |                                      |
| **Complex Features**        | Simple and lightweight; avoids complex processing.                                                        | Includes features like stats, notifications, importing and exporting.   |

You can also play with the demo: https://demo.dawarich.app/

**Comment by gene1wood** [comment]
Hmm, ok.

It looks like 
* OwnTracks Recorder does [support reverse Geocoding using OpenCage](https://owntracks.org/booklet/other/opencage/)
* And it has [heatmap visualization](https://github.com/owntracks/frontend?tab=readme-ov-file#features)
* And it [imports Google Timeline data](https://github.com/owntracks/recorder/tree/master/contrib/google-import) though apparently [this method is better](https://www.technowizardry.net/2024/01/migrating-from-google-location-history-to-owntracks/)

@Freika , was the data that you put in the table with ChatGPT, stuff you came up with or is ChatGPT just confused/mistaken?

**Reply by Freika** [reply]
I made some edits to the table, but also I might have forgotten or didn't know about the existence of some of the features that Owntracks has :) It's been a while since I was using it, and I definitely didn't use reverse geocoding with Owntracks

**Reply by Freika** [reply]
@gene1wood updated the table a bit, thanks :)

**Comment by gene1wood** [comment]
@Freika Maybe a better question I should ask (that isn't answerable by ChatGPT) is, what motivated you to make dawarich given the other solutions (like Owntracks Recorder) that exist? What was missing from other solutions that made you want to create dawarich?

**Reply by Freika** [reply, marked as answer]
That's a good question :)

I used Owntracks for about 6 months and it was good but pretty basic experience.

I wanted to be able to visualize my location history in a way I find convenient. Also, I wanted to have more insights of places I visited, distances I made, see what countries and cities I visited in given timespan, and so on. And also I love building stuff. I think that would be my answer :)
